### PR TITLE
[WEBSITE-1225] Basic regex to prevent xss on search

### PR DESCRIPTION
**Pull request by UsmanAfzal**

### CODE REVIEW

File: src/elements/mixins.pug [hunk 1]
```diff
@@ -17,6 +17,6 @@ mixin image(src, alt)
   img(src=src alt=alt)&attributes(attributes)
 
 //- Inputs
-mixin input(label, id, maxlength, regex)
+mixin input(label, id, type, maxlength, regex, title)
   label(for=id)= label
-  input(id=id, name=id, type='text' maxlength=maxlength pattern=regex)
\ No newline at end of file
+  input(id=id, name=id, type=type maxlength=maxlength pattern=regex title=title)
```
Comment (jakemulley) [inline src/elements/mixins.pug]: What if there isn't a pattern applicable?

File: src/templates/prototypes/forms.pug [hunk 2]
```diff
@@ -21,7 +19,7 @@ block content
 
           form(action='postcode')
             .form-group
-              +input('Search by postcode, for example; SW1A 0AA', 'postcode_main', 8, '[0-9a-zA-Z ]{5,}')
+              +input('Search by postcode, for example; SW1A 0AA', 'postcode_main', 'text', 8, '[0-9a-zA-Z ]{5,}', '- Special characters (!@£$%) are not permitted')
```
Comment (jakemulley) [inline src/templates/prototypes/forms.pug]: Confirm this message with content - seems nonsensical to include that message when you can only enter numbers and letters anyway

File: src/elements/mixins.pug (hunk 1 above)
Comment (UsmanAfzal) [inline src/elements/mixins.pug]: When we come across an input without a required pattern we can create that variation

File: src/templates/prototypes/forms.pug (hunk 2 above)
Comment (UsmanAfzal) [inline src/templates/prototypes/forms.pug]: Already in a ticket for content to amend.

Currently all characters can get posted. Makes sense to have an initial preventative measure for unnecessary posts

Comment (jakemulley) [inline src/templates/prototypes/forms.pug]: Sure, which is fine, but that message really doesn't make sense. What if I enter an asterisk (*), or a question mark(?). Listing only `(!@£$%)` wouldn't make sense. It'd make sense to change this to "Only alphanumeric characters (A-Z or 0-9) allowed" at the moment.

**Pull request merged**